pkg/kubelet/dockertools/manager.go

```diff
@@ -924,7 +929,7 @@ func (dm *DockerManager) podInfraContainerChanged(pod *api.Pod, podInfraContaine
 		networkMode = dockerPodInfraContainer.HostConfig.NetworkMode
 	}
 	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.HostNetwork {
```

Comment:
Also, this doesn't cover the case where the podSpec has changed to !HostNetwork but docker is still on host network; is that a valid case?  Not that the original code covered that case either...